walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,121 @@
+
+        // Array of queries represenitng times to meet.
+        List<TimeRange> query = new ArrayList<>();
+        
+        // Integer represenitng the number of events.
+        int numberOfEvents = events.size();
+        
+        // Integer representing the Start of a work day.
+        int timeOfStart = START_OF_DAY;
+
+        // Integer representing the End of a work day.
+        int timeOfEnd = END_OF_DAY;
+        
+        // Integer represenig the duration of a meeting request.
+        int desiredDuration = (int) request.getDuration();
+        
+        //  Event counter used whilst iterating.
+        int eventCounter = 0;
+        
+        int event_start = 0;
+        int event_end = 0;
+
+        // Set of stirngs representing unique employees set to attend this meeting.
+        Set<String> event_attendees = null;
+
+        // Collection of string reprenting who is to attend the requedted meeting
+        Collection<String> request_attendees = request.getAttendees();
+        
+        // Boolean used to indictae whter or not the attendee should be ignored.
+        boolean shouldIgnore = true;
+
+        // Check if duration of event is longer than a whole day and if so return empty.
+        if ((int)request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return query;
+        }
+
+        // Check if there are any events happening at all if so return whole day query.
+        if (numberOfEvents == 0) {
+            // Adds a 24 hour time range to query.
+            query.add(TimeRange.WHOLE_DAY);
+            return query;
+        }
+        // Iterating throguh input event collection. 
+        for (Event event : events) {
+
+            // Collection of the pople who are to attend meeting.
+            attendees = event.getAttendees();
+
+            //Ignore event only if attendees of this event are not needed in requested meeting
+            for (String attendee : attendees) {
+                // If this attendee is required
+                if (request_attendees.contains(attendee)) {
+                    // You shouldn't ignore this attendee. 
+                    shouldIgnore = false;
+                    break;
+                } 
+                // If at the last event.
+                else if (eventCounter == numberOfEvents-1) {
+                    query.add(TimeRange.fromStartEnd(timeOfStart, timeOfEnd, true));
+                }
+            }
+            // if the attendee is mandatory.
+            if (!shouldIgnore) {
+                
+                // Fetch event start time.
+                event_start = event.getWhen().start();
+                // Fetch event end time.
+                event_end = event.getWhen().end();
+
+                if (event_start == timeOfStart) {
+                    timeOfStart = event_end;
+                }
+
+                // Check for overlapping/nested events
+                if (event_start < timeOfStart) {
```

Comment:
`timeOfStart` might be updated after the previous `if`. I think it's not intended, right? So you might need a temporary variable to store the value of `timeOfStart` for the next iteration.